ocs_ci/ocs/dr/dr_workload.py

```diff
@@ -1647,25 +1647,38 @@ def __init__(self, **kwargs):
             "discovered_apps_pod_selector_value"
         )
 
-    def deploy_workload(self):
+    def deploy_workload(self, shared_drpc_protection=False, dr_protect=True):
         """
 
+        Args:
+            shared_drpc_protection (bool): False by default, another workload in an existing namespace
+                            will share the DRPC and DR protect it using Shared Protection type via UI
+            dr_protect (bool): True by default where workload will be DR protected via CLI,
+                                else test case should handle it (via UI)
+
         Deployment specific to CNV workload for Discovered/Imperative Apps
 
         """
         self._deploy_prereqs()
         for cluster in get_non_acm_cluster_config():
             config.switch_ctx(cluster.MULTICLUSTER["multicluster_index"])
-            self.create_namespace()
-        self.manage_dr_vm_secrets()
+            if not shared_drpc_protection:
+                self.create_namespace()
+            else:
+                # Shared=False means namespace exists, so skip creation
+                log.info(
+                    f"Namespace in use: {self.workload_namespace} for Shared Protection type"
+                )
+        self.manage_dr_vm_secrets(shared_drpc_protection=shared_drpc_protection)
```

Comment:
@am-agrawa I feel, we can call manage_dr_vm_secrets under `if not shared_drpc_protection:` in line no 1666 and remove any changes made to manage_dr_vm_secrets. so that we can avoid unneccesary code in manage_dr_vm_secrets funtion. Anyways we call this function with shared_drpc_protection as True during the creation of second vm, so automatically the execution will go to line no 1667